csrc/ir/utils.h

```diff
@@ -173,16 +173,15 @@ TORCH_CUDA_CU_API Expr* replaceValInExpr(
     Val* reference,
     Val* substitute);
 
-//! Replace Vals in an index Val as specified by replacement_map while
-//! cloning the given index Val. The index val is assumed to represent
-//! a tensor index consisting of Ints and arithmetic expressions.
+//! Replace Vals in the definition of the given Val as specified by
+//! replacement_map while cloning the given Val.
 //!
 //! This is similar to replaceValInExpr but is different as Vals are
 //! cloned such that no other exprs using the same leaf Vals are not
 //! modified. TODO: Consider cleaning up the multiple replacement
 //! routines.
-Val* replaceValInIndexVal(
-    Val* index,
+Val* replaceValInDef(
```

Comment:
nit: the naming might be a little confusing as the replacement is not necessarily done in the definition. Maybe, something like `replaceValRecursively`?

Reply:
changed